### ISSUE TITLE
[CUSTOM CODE views] Colors and symbols of the legend are not integrated into the map composer

### ISSUE THREAD

**Issue by antobenve** [unep-grid/mapx#1006]
As shown in the example below, none of the colored polygons that are present in  the legend of the selected layer are displayed in the map composer. This applies to all custom code views as far as I could see.
![image](https://github.com/user-attachments/assets/ae83cc75-ff85-45de-a4f9-5bc5eee5a160)

**Comment by antobenve** [comment]
[The one in the screenshot for instance](https://app.mapx.org/?language=en&project=MX-KVV-ANS-WBN-QXR-S71&viewsListFlatMode=false&viewsOpen=MX-KG1JB-OI13A-6RHKN&viewsListFilterActivated=true):
```

/**
 * Custom method for custom view in MapX
 * Parameters for onInit and onClose function ;
 * @param {Object} o Options
 * @param {Object} o.view Map-x view object
 * @param {Object} o.map mapbox-gl map object
 * @param {String} o.idView If of the view
 * @param {String} o.idSource Id of the source
 * @param {Element} o.elLegend Element containing the legend
 *
 */
return {
    onClose: function(o) {
        /**
         * Remove source
         */
        o.onRemove();

    },
    onInit: function(o) {
        const h = mx.helpers;
        const el = h.el;

        var categories = [{
                label: "Tree cover",
                color: "rgb(0,100,0)"
            },
            {
                label: "Shrubland",
                color: "rgb(255, 187, 34)"
            },
            {
                label: "Grassland",
                color: "rgb(255, 255, 76)"
            },
            {
                label: "Cropland",
                color: "rgb(240, 150, 255)"
            },
            {
                label: "Built-up",
                color: "rgb(250, 0, 0)"
            },
            {
                label: "Bare/sparse vegetation",
                color: "rgb(180, 180, 180)"
            },
            {
                label: "Snow and Ice",
                color: "rgb(240, 240, 240)"
            },
            {
                label: "Permanent water bodies",
                color: "rgb(0, 100, 200)"
            },
            {
                label: "Herbaceous wetland",
                color: "rgb(0, 150, 160)"
            },
            {
                label: "Mangroves",
                color: "rgb(0, 207, 117)"
            },
            {
                label: "Moss and lichen",
                color: "rgb(250, 230, 160)"
            }
        ];
        
        setLegend(categories);

        let url = getUrl('WORLDCOVER_2021_MAP');

        /**
         * Set custom source
         */
        o.source = {
            type: 'raster',
            tiles: [url],
            tileSize: 512
        };

        /**
         * Set custom layer
         */
        o.layer = {
            id: o.idView,
            type: "raster",
            source: o.idSource,
            paint: {}
        };

        /**
         * Set layer and source
         */
        o.map.addSource(o.idSource, o.source);
        o.map.addLayer(o.layer, mx.settings.layerBefore);


        /**
         * Helpers
         */
        function getUrl(layer) {
            var url = 'https://services.terrascope.be/wms/v2?';
            out = mx.helpers.objToParams({
                bbox: 'bboxcode',
                service: 'WMS',
                version: '1.1.1',
                styles: '',
                request: 'getMap',
                zindex: '10',
                srs: 'EPSG:3857',
                layers: layer,
                format: 'image/png',
                transparent: 'true',
                height: '512',
                width: '512',
                exceptions: 'application:Fvnd.ogc.se_blank'
            });
            out = out.replace('bboxcode', '{bbox-epsg-3857}');
            out = url + out;
            return out;
        }

        function setLegend(categories) {
            if (h.isElement(o.elLegend)) {

                var elLegendContent = el("div", {
                        class: ['mx-legend-box'],
                        style: {
                            display: "flex",
                            flexDirection: "column"
                        }
                    },
                    el('div', 'Land Cover Class', {
                        style: {
                            paddingLeft: '10px',
                            marginBottom: '10px'
                        }
                    }),
                    categories.map(e => {
                        return el('div', {
                                style: {
                                    display: 'flex',
                                    flexDirection: 'row',
                                    justifyContent: 'flex-start',
                                    alignItems: 'center'
                                }
                            },
                            el('span', {
                                style: {
                                    width: '20px',
                                    height: '20px',
                                    marginRight: '5px',
                                    background: e.color
                                }
                            }),
                            el('span', e.label));
                    }));
                o.elLegend.appendChild(elLegendContent);
            }
        }

        /**
         * Remove added layers
         */
        o.onRemove = function() {
            var layer = o.map.getLayer(o.idView);
            var source = o.map.getSource(o.idSource);
            if (layer) {
                o.map.removeLayer(o.idView);
            }
            if (source) {
                o.map.removeSource(o.idSource);
            }
        };

    }
};
```

**Comment by fxi** [comment]
ok. the issue was specific to this view : it uses empty `<span>` to style the colors of the legend. If other cases occurs, please report them, with a reproducible example : the MapComposer could alter many things : removing events, id, style,  and add editable parameter for the live editor.

**Comment by antobenve** [comment]
Well, the issue is specific to all views I could test but probably they all used the same method.
See [Cartagena](https://app.mapx.org/?project=MX-L2W-HWZ-RIC-LM1-Y0V&viewsOpen=MX-33ZSA-2J004-NCVDF&viewsListFilterActivated=true&theme=classic_light&language=en) or more simply [Base data](https://app.mapx.org/?project=MX-KVV-ANS-WBN-QXR-S71&viewsOpen=MX-33ZSA-2J004-NCVDF&viewsListFilterActivated=true&theme=classic_light&language=en). I can't list them all.

**Comment by fxi** [comment]
Yes, report them if needed, after the patch is available, probably in `1.13.10-alpha.1`, if it does not work for them. 
If they use the same method, then it should be fixed by d11ba67 
Note that this issue is marked as 'done' in the planner.
